### PR TITLE
- Migrated PROP-PTX and PTX-GIS to PA_PARCEL, PAGIS

### PR DESCRIPTION
**Pull request by RamalingamThangavel**
- Migrated
                     - PROP-PTX to PA_PARCEL
                     - PTX-GIS to PAGIS

- Fixed the unit tests and tested in dev. env.

### CODE REVIEW

File: src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy [hunk 1]
```diff
@@ -207,39 +214,42 @@ class PropertyInfoService {
...
+		else if (dataFromLayers['MDC.PaParcel']) {
+			propertyType = getPropertyType(dataFromLayers, 'MDC.PaParcel')
 		}
 
 		return propertyType
 	}
 
 	/**
+	 * Get Property type.
+	 *
```
Comment (fiallega) [inline src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy:229]: I would have a lengthy explanation on how you calculate each value, like for example MULTI

File: src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy [hunk 2]
```diff
@@ -207,39 +214,42 @@ class PropertyInfoService {
...
+		PropertyType propertyType = PropertyType.UNDEFINED;
+		String isCondo = dataFromLayers[layerName].attributes?.CONDO_FLAG ?: dataFromLayers[layerName].CONDO_FLAG
+		String parentFolioNumber = dataFromLayers[layerName].attributes?.PARENT_FOLIO ?: dataFromLayers[layerName].PARENT_FOLIO
+		String folioNumber = dataFromLayers[layerName].attributes?.FOLIO ?: dataFromLayers[layerName].FOLIO
+		String referenceOnlyFlag = dataFromLayers[layerName].attributes?.REFERENCE_ONLY_FLAG ?: dataFromLayers[layerName].REFERENCE_ONLY_FLAG
+
+		if (isCondo.toBoolean()) {
+			propertyType = PropertyType.CONDO
+		} else if ( (parentFolioNumber != folioNumber && folioNumber.endsWith("0001")) && referenceOnlyFlag.toBoolean()) {
+			propertyType = PropertyType.MULTI
```
Comment (fiallega) [inline src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy]: if there are all && why do you need parentheses ((x && y) && zz) = (x && y && z)

File: src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy [hunk 3]
```diff
@@ -244,38 +244,40 @@ class Esri10AddressServiceTest {
...
+		assert address.addressApproximation == false
+		assert address.propertyInfo.parcelFolioNumber == '0341170040220'
 		assert address.propertyInfo.propertyType == 'UNDEFINED'
 		assert address.streetsMaintenance ==  [
-			[streetName:'RIVIERA DR', maintenanceCode:'CI', maintenanceCodeAlias:'City maintained']
+			[streetName:'UNIVERSITY DR', maintenanceCode:'CI', maintenanceCodeAlias:'City maintained']
 		]
 		assert address.streetsLightMaintenance == [
-			[streetName:'RIVIERA DR',  maintenanceCode:'LIGHT', maintenanceCodeAlias:'FPL - (800) 468-8243']
+				[streetName:'ANASTASIA AVE', maintenanceCode:'LIGHT', maintenanceCodeAlias:'FPL - (800) 468-8243'], [streetName:'UNIVERSITY DR', maintenanceCode:'LIGHT', maintenanceCodeAlias:'FPL - (800) 468-8243']
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy:265]: Can we look at this. I am not sure why do we have two entries in the array

File: src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy [hunk 4]
```diff
@@ -378,19 +380,19 @@ class Esri10AddressServiceTest {
...
-			assert municipalityId == 3
-			assert location.x == 898029
-			assert location.y ==  512098.0
-			assert addressApproximation == true
-			assert propertyInfo.parcelFolioNumber == '0341170040650'
+			assert municipalityId == 03
+			assert location.x == 898469.263
+			assert location.y ==  512484.987
+			assert addressApproximation == false
+			assert ( propertyInfo.parcelFolioNumber == '0341170040650' || propertyInfo.parcelFolioNumber == '0341170040220')
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy]: Why two values for the parcelFolioNumber ?

File: src/test/groovy/org/sharegov/mdcgis/FeatureServiceTests.groovy [hunk 5]
```diff
@@ -259,8 +259,8 @@ class FeatureServiceTests {
 		Map attributes = featureService.propertyAttributesByFolio("3059010240130")
 		attributes.with {
 			assert FOLIO == "3059010240130"
-			assert PARENT == "3059010240130"
-			assert PTXADDRESS == "11826 SW 97 ST"
+			assert PARENT_FOLIO == null
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/FeatureServiceTests.groovy:262]: How is that work, why is the parent folio null ?

File: src/test/groovy/org/sharegov/mdcgis/GisConfigFactoryBeanTests.groovy [hunk 6]
```diff
@@ -47,9 +47,9 @@ class GisConfigFactoryBeanTests {
 	
 	@Test
 	public void testLoadLayers(){
-		assert config.layers.size() == 41
+		assert config.layers.size() == 42	//There is changes in the layers count???? Jorge...confirm
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/GisConfigFactoryBeanTests.groovy:50]: I believe this is ok.

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy [hunk 7]
```diff
@@ -57,14 +57,9 @@ class PropertyInfoServiceTests {
...
-			assert parcelInfoLegal1 == "GLEN COVE WEST SEC 1"
-			assert parcelInfoLegal2 == "PB 118-30"
-			assert parcelInfoLegal3 == "LOT 13 BKL 1"
-			assert parcelInfoLegal4 == "LOT SIZE 8800 SQ FT"
-			assert parcelInfoLegal5 == "OR 17101-5181 0296 1"
-			assert parcelInfoLegal6 == "OR 17101-5181 0296 00"
+			assert parcelInfoLegal == "GLEN COVE WEST SEC 1 PB 118-30 LOT 13 BKL 1 LOT SIZE 8800 SQ FT OR 17101-5181 0296 1"
 			assert parcelInfoPtxAddress == "11826 SW 97 ST"
-			assert parcelInfoAddress == "11826 SW 97TH ST"
+			assert parcelInfoAddress == "11826 SW 97 ST"
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy:62]: That surprises me, that 97 is not 97TH. What is parcelInfoPtxAddress vs parcelInfoAddress ?

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy [hunk 8]
```diff
@@ -95,13 +85,8 @@ class PropertyInfoServiceTests {
...
 		propertyInfo.with {
 			assert parcelInfoPtxAddress == ""
-			assert parcelInfoAddress == ""
-			assert parcelInfoLegal1 == "HALLANDALE PARK NO 8 PB 20-49"
-			assert parcelInfoLegal2 == "LOT 15 BLK 18 & S1/2 OF ALLEY & N"
-			assert parcelInfoLegal3 == "NE 206 ST LYG N & S & ADJ CLOSED"
-			assert parcelInfoLegal4 == "PER R-2006-61"
-			assert parcelInfoLegal5 == "LOT SIZE 3966 SQ FT"
-			assert parcelInfoLegal6 == "FAU 30 1234 006 4600"
+			assert parcelInfoAddress == null
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy:88]: why is this null and not "" ?

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy [hunk 9]
```diff
@@ -124,12 +109,21 @@ class PropertyInfoServiceTests {
...
 	public void testGetCondoPropertyInfo_NoData(){
-		Map data = [:]
-		assert propertyInfoService.getCondoPropertyInfo(data) == null
-		data = null
-		assert propertyInfoService.getCondoPropertyInfo(data) == null
+		try {
+			Map data = [:]
+			def result = propertyInfoService.getCondoPropertyInfo(data)
+			assert false
+
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy]: not sure if  I understand this. assert false but then you seem to have more code afterwards

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy [hunk 10]
```diff
@@ -145,31 +139,35 @@ class PropertyInfoServiceTests {
 
 	@Test
 	public void testGetCleanPrpertyInfoByFolio_ConvertXYtoDoubleWith3Decimals(){
+		/*Do we need this ????!!!!!!!
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy]: lets look at this together. That is the way we mock information.

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy [hunk 11]
```diff
@@ -219,21 +217,22 @@ class PropertyInfoServiceTests {
...
 	}
 
+
 	@Test
 	public void testGetStreetZipUnitByFolio_Condo(){
 		Map data = propertyInfoService.getStreetZipUnitByFolio("0232341690630")
-		assert data.zip == 33139
-		assert data.street == '2001 MERIDIAN AVE'
+		assert data.zip == '33139-1503'
+		assert data.street == '2001 MERIDIAN AVE 317'
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy]: so, the data.street includes unit ? is that ok ?

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy [hunk 12]
```diff
@@ -349,20 +349,20 @@ class PropertyInfoServiceTests {
...
 		assert units.size() == 113
 
 	}
-	
+
 	@Test
 	void testGetAddressForMultiAddressBuilding(){
 		String address = propertyInfoService.getAddressForMultiAddressBuilding("3049331130001") 
-		assert  address == "8255 SW 152ND AVE"
+		assert  address == "8215 SW 152ND AVE"
```
Comment (fiallega) [inline src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy:358]: How are we dealing with AddressForMultiAddressBuilding ?

File: src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy (hunk 2 above)
Comment (RamalingamThangavel) [reply src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy]: done

Comment (fiallega) [reply src/main/groovy/org/sharegov/mdcgis/PropertyInfoService.groovy]: ok.

File: src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy (hunk 3 above)
Comment (fiallega) [reply src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy:265]: ok.

File: src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy (hunk 4 above)
Comment (fiallega) [reply src/test/groovy/org/sharegov/mdcgis/Esri10AddressServiceTest.groovy]: ok

File: src/test/groovy/org/sharegov/mdcgis/FeatureServiceTests.groovy (hunk 5 above)
Comment (fiallega) [reply src/test/groovy/org/sharegov/mdcgis/FeatureServiceTests.groovy:262]: ok

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy (hunk 7 above)
Comment (fiallega) [reply src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy:62]: ok

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy (hunk 8 above)
Comment (fiallega) [reply src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy:88]: ok

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy (hunk 9 above)
Comment (fiallega) [reply src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy]: ok

File: src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy (hunk 11 above)
Comment (fiallega) [reply src/test/groovy/org/sharegov/mdcgis/PropertyInfoServiceTests.groovy]: ok

**Pull request merged**